pilot/pkg/model/sidecar.go

```diff
@@ -903,31 +906,67 @@ func (sc *SidecarScope) appendSidecarServices(servicesAdded map[host.Name]sideca
 			return
 		}
 
+		if !canMergeServices(existing, s) {
+			log.Debugf("Service %s/%s from registry %s ignored by %s/%s/%s", s.Attributes.Namespace, s.Hostname, s.Attributes.ServiceRegistry,
+				existing.Attributes.Namespace, existing.Hostname, existing.Attributes.ServiceRegistry)
+			return
+		}
+
 		// we merge ports for services both defined by ServiceEntry in same namespace
-		if existing.Attributes.Namespace == s.Attributes.Namespace {
-			// merge the ports to service when each listener generates partial service
-			// we only merge if the found service is in the same namespace as the one we're trying to add
-			copied := foundSvc.svc.DeepCopy()
-			for _, p := range s.Ports {
-				found := false
-				for _, osp := range copied.Ports {
-					if p.Port == osp.Port {
-						found = true
-						break
-					}
-				}
-				if !found {
-					copied.Ports = append(copied.Ports, p)
+
+		// merge the ports to service when each listener generates partial service
+		// we only merge if the found service is in the same namespace as the one we're trying to add
+		copied := foundSvc.svc.DeepCopy()
+		for _, p := range s.Ports {
+			found := false
+			for _, osp := range copied.Ports {
+				if p.Port == osp.Port {
+					found = true
+					break
 				}
 			}
-			// replace service in slice
-			sc.services[foundSvc.index] = copied
-			// Update index as well, so that future reads will merge into the new service
-			foundSvc.svc = copied
-			servicesAdded[foundSvc.svc.Hostname] = foundSvc
-			sc.servicesByHostname[s.Hostname] = s
+			if !found {
+				copied.Ports = append(copied.Ports, p)
+			}
 		}
+		// replace service in slice
+		sc.services[foundSvc.index] = copied
+		// Update index as well, so that future reads will merge into the new service
+		foundSvc.svc = copied
+		servicesAdded[foundSvc.svc.Hostname] = foundSvc
+		sc.servicesByHostname[s.Hostname] = s
+	}
+}
+
+func canMergeServices(s1, s2 *Service) bool {
+	if s1.Attributes.Namespace != s2.Attributes.Namespace {
+		return false
+	}
+
+	// Hostname has been compared
```

Comment:
What does this comment mean?

Reply:
It is guaranteed when  we call this function, the services already have same hostname

Reply:
can you please update the comment to reflect the same?

Reply:
OK, it is a little confusing now